doc/api/buffer.md

```diff
@@ -1150,6 +1150,34 @@ console.log(utf16Buffer.indexOf('\u03a3', 0, 'ucs2'));
 console.log(utf16Buffer.indexOf('\u03a3', -4, 'ucs2'));
 ```
 
+Edge cases:
+
+If `value` is not a string, number, or Buffer, this method will throw a
+`TypeError`. If `value` is a number, it will be coerced to a valid byte value,
+an integer between 0 and 255.
+
+If `byteOffset` is not a number, it will be coerced to a number. Any arguments
+that coerce to `NaN` or 0, like `{}`, `[]`, `null` or `undefined`, will search
+the whole buffer. This behavior matches [`String.indexOf()`].
+
+Edge case examples:
+
+```js
+const b = Buffer.from('abcdef')
+
+// Passing a value that's a number, but not a valid byte
+// Prints: 2, equivalent to searching for 99 or 'c'
```

Comment:
"Prints" is not accurate. These don't print anything except in the REPL. I believe the way we handle this in other contexts is like this:

```js
b.indexOf(99.9); // 2
```

Also, note that we use semi-colons in the example code throughout the docs. Please do the same here. Thanks!